File: SyscallTest.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    const char *filename = "example.txt";
    const char *content = "Hello from C!";

    // Write to file
    FILE *file = fopen(filename, "w"); //Uses Ubuntu's "open" syscall
    if (file == NULL) {
        perror("Error opening file for writing");
        return 1;
    }
    fprintf(file, "%s", content); //Uses Ubuntu's "write" syscall
    fclose(file);
    printf("Written to file: %s\n", filename);

    // Read from file
    file = fopen(filename, "r"); //Uses Ubuntu's "open" syscall
    if (file == NULL) {
        perror("Error opening file for reading");
        return 1;
    }

    char buffer[100];
    printf("Reading from file:\n");
    while (fgets(buffer, sizeof(buffer), file)) { //Uses Ubuntu's "read" syscall
        printf("%s", buffer);
    }

    fclose(file);
    return 0;
}

File: SyscallTest.java
import java.io.*;

public class SyscallTest {
    public static void main(String[] args) {
        String filename = "javafile.txt";
        String content = "Hello from Java!";

        // Write to file
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) { //Calls Windows' CreateFile syscall
            writer.write(content); //Calls Windows' WriteFile syscall
            System.out.println("Written to file: " + filename);
        } catch (IOException e) {
            e.printStackTrace();
        }

        // Read from file
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) { //Calls Windows' ReadFile syscall
            String line;
            System.out.println("Reading from file:");
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
